kafka-event-listener/src/main/java/org/hypertrace/core/kafka/event/listener/KafkaLiveEventListener.java

```diff
@@ -0,0 +1,93 @@
+package org.hypertrace.core.kafka.event.listener;
+
+import com.typesafe.config.Config;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.Future;
+import java.util.concurrent.TimeUnit;
+import java.util.function.BiConsumer;
+import org.apache.kafka.clients.consumer.Consumer;
+import org.apache.kafka.common.serialization.Deserializer;
+
+/**
+ * KafkaLiveEventListener consumes events produced to a single Kafka Topic from its initialisation
+ * and on every message invokes provided callbacks. The thread safety of callback method must be
+ * ensured by provider. It is important to note that there is no guarantee that all messages on
+ * topic are consumed by this listener as every time we create a listener we consume from latest
+ * offsets by design.
+ *
+ * <p>Typical usage of this listener is to back the remote caches to have lower latency of refresh
+ * by generating respective information on kafka topics
+ */
+public class KafkaLiveEventListener<K, V> implements AutoCloseable {
+  private final Future<Void> kafkaLiveEventListenerCallableFuture;
+  private final ExecutorService executorService;
+  private final boolean cleanupExecutor;
+
+  private KafkaLiveEventListener(
+      KafkaLiveEventListenerCallable<K, V> kafkaLiveEventListenerCallable,
+      ExecutorService executorService,
+      boolean cleanupExecutor) {
+    this.executorService = executorService;
+    this.cleanupExecutor = cleanupExecutor;
+    this.kafkaLiveEventListenerCallableFuture =
+        executorService.submit(kafkaLiveEventListenerCallable);
+  }
+
+  @Override
+  public void close() throws Exception {
+    kafkaLiveEventListenerCallableFuture.cancel(true);
+    if (cleanupExecutor) {
+      executorService.awaitTermination(10, TimeUnit.SECONDS);
```

Comment:
Await Termination alone is not sufficient. It won't initiate the shutdown on its own
It has to be called after shutdown or shutdownNow.